bpf/lib/nodeport.h

```diff
@@ -693,14 +693,24 @@ int tail_nodeport_nat_ingress_ipv6(struct __ctx_buff *ctx)
 
 	ret = snat_v6_rev_nat(ctx, &target);
 	if (IS_ERR(ret)) {
-		/* In case of no mapping, recircle back to main path. SNAT is very
-		 * expensive in terms of instructions (since we don't have BPF to
-		 * BPF calls as we use tail calls) and complexity, hence this is
-		 * done inside a tail call here.
-		 */
-		ctx_skip_nodeport_set(ctx);
-		ep_tail_call(ctx, CILIUM_CALL_IPV6_FROM_NETDEV);
-		ret = DROP_MISSED_TAIL_CALL;
+		if (ret == NAT_PUNT_TO_STACK ||
+		    /* DROP_NAT_NO_MAPPING and DROP_NAT_UNSUPP_PROTO
+		     * are both unwanted behavior in a revNAT
+		     * context. Let's continue to passing it up to the
+		     * host, and revisiting this later if needed.
+		     */
+		    ret == DROP_NAT_NO_MAPPING || ret == DROP_NAT_UNSUPP_PROTO) {
```

Comment:
Instead of the additional check for `DROP_NAT_UNSUPP_PROTO`, how about we change https://github.com/cilium/cilium/blob/0d7406af7bf05e1c0c49c098a3b4e025d927c3e7/bpf/lib/nat.h#L1797 to return `NAT_PUNT_TO_STACK` instead - same as a few lines below for any unknown L4 protocol?

Two reasons why I think that's preferable
- we're not exposing ourselves to some deeper-down code path that returns `DROP_NAT_UNSUPP_PROTO`, and where we *want* to treat it as hard error
- it's slightly less code :)

Could you please spell out clearly in the patch description that current code is silently ignoring actual errors, from eg. when the revSNAT code tries to rewrite the packet headers? We might want to backport this as a fix.
It's also probably best to explain why DROP_NAT_NO_MAPPING can be ignored for now.

Reply:
It's all done